vmtools.py

```diff
@@ -158,6 +160,27 @@ def delete_old_backups(api, config, vm_name):
                         logger.debug("Delete old backup (%s) in progress ..." % vm_name_export)
                         time.sleep(config.get_timeout())
 
+    @staticmethod
+    def delete_old_backups_by_number(api, config, vm_name):
+        """
+        Delete old backups from the export domain by number of requested
+        :param api: ovirtsdk api
+        :param config: Configuration
+        """
+        vm_search_regexp = ("%s%s*" % (vm_name, config.get_vm_middle())).encode('ascii', 'ignore')
+        exported_vms = api.storagedomains.get(config.get_export_domain()).vms.list(name=vm_search_regexp)
+        exported_vms.sort(key=lambda x: x.get_creation_time())
+        while len(exported_vms) > config.get_backup_keep_count_by_number():
+            i = exported_vms.pop(0)
+            vm_name_export = str(i.get_name())
+            logger.info("Backup deletion started for backup: %s", vm_name_export)
+            if not config.get_dry_run():
+                i.delete()
+            while vm_name_export in [vm.name for vm in api.storagedomains.get(config.get_export_domain()).vms.list()]:
```

Comment:
api.storagedomains.get(config.get_export_domain()) does not need to be within the while statement.
I assume it just causes additional REST API queries.

Reply:
done